consensus/clique/clique.go

```diff
@@ -678,10 +686,15 @@ func (c *Clique) CalcDifficulty(chain consensus.ChainHeaderReader, time uint64,
 	c.lock.RLock()
 	signer := c.signer
 	c.lock.RUnlock()
-	return calcDifficulty(snap, signer)
+	return c.calcDifficulty(snap, signer)
 }
 
-func calcDifficulty(snap *Snapshot, signer common.Address) *big.Int {
+func (c *Clique) calcDifficulty(snap *Snapshot, signer common.Address) *big.Int {
+	if c.config.ShadowForkHeight > 0 && snap.Number >= c.config.ShadowForkHeight {
+		// if we are past shadow fork point, set a high difficulty so that forked nodes don't switch to main chain
```

Comment:
Does this mean that nodes will prefer higher difficulty, even with Clique?

Reply:
Yeah, Clique still uses TD as its fork-choice algorithm. Normally, if a node is spawned with `--shadowfork` flag it won't switch back to the original chain. But if you accidentially omit the flag (which happened to me), it might switch to the original chain and result in data loss. So I wanted to make sure reverting a shadowforked chain was always intentional.

Reply:
But after thinking out loud, it might make sense to take the risk and make the shadowforked chain have a lower difficulty as well. That would make sure that main chain nodes never attempt to switch over to it, making them less affected from our shadowforks. Currently they try to switch over to shadowfork periodically and drop the connection to shadowfork node after they realize the chain is no good.

Reply:
> drop the connection to shadowfork node after they realize the chain is no good.

Yes, that's what I expected. But won't this prevent us from receiving post-shadowfork transactions?

Reply:
Resolving, discussed elsewhere.